pkg/util/kube/utils.go

```diff
@@ -97,6 +103,9 @@ func EnsureNamespaceExistsAndIsReady(namespace *corev1api.Namespace, client core
 
 	// err will be set if we timed out or encountered issues retrieving the namespace,
 	if err != nil {
+		if terminatingNamespace {
+			return false, nsCreated, errors.Wrapf(err, "timed out waiting for terminating namespace %s to disappear before restoring", namespace.Name)
+		}
```

Comment:
as this call is used during restore as well, and if such error occur for the resource during restore, can we also add this to restore warnings list?

Reply:
Mention #7516

Reply:
Is warning preferred over restore error?

Reply:
Also per PR title, the intent is that the error are visible in restore error.

Reply:
What changes do you see needed to be made to the PR?

Reply:
Sorry on the late reply, I was awk.
Yes, overall its good to have this error that you added and its visible during restore

Reply:
For reducing the time to address the issue #7516 we need to look at other options.